test/evergreen/configure_combinations.sh

```diff
@@ -1,59 +1,84 @@
 #!/bin/bash
 
+: ${CMAKE_BIN:=cmake}
+
+for i in "$@"; do
+  case $i in
+    -g=*|--generator=*)
+      GENERATOR="${i#*=}"
+      shift # past argument=value
+      ;;
+    -j=*|--parallel=*)
+      PARALLEL="-j ${i#*=}"
+      shift # past argument=value
+      ;;
+    *)
+      # unknown option
+      ;;
+  esac
+done
+
+if [ -z "${GENERATOR}" ]; then
+    GENERATOR="Unix Makefiles"
+fi
+if [ "$GENERATOR" != "Ninja" ] && [ "$GENERATOR" != "Unix Makefiles" ]; then
+    echo "Invalid build generator: $GENERATOR. Valid options 'Ninja', 'Unix Makefiles'"
+fi
+
+if [ "$GENERATOR" == "Unix Makefiles" ]; then
+    GENERATOR=$(echo $GENERATOR | sed -e 's/ /\\ /')
+    GENERATOR_CMD="make"
+else
+    GENERATOR_CMD="ninja"
+fi
+
 cd $(git rev-parse --show-toplevel)
 echo `pwd`
-sh build_posix/reconf
 
 curdir=`pwd`
 
-flags="CFLAGS=\"-Werror -Wall -Wextra -Waddress -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wformat-security -Wformat-nonliteral -Wformat=2 -Wmissing-declarations -Wmissing-field-initializers -Wmissing-prototypes -Wnested-externs -Wno-unused-parameter -Wpointer-arith -Wredundant-decls -Wshadow -Wundef -Wunused -Wwrite-strings -O -fno-strict-aliasing -Wuninitialized\"
-CC=clang CFLAGS=\"-Wall -Werror -Qunused-arguments -Wno-self-assign -Wno-parentheses-equality -Wno-array-bounds\""
+flags="-DCMAKE_TOOLCHAIN_FILE=$curdir/cmake/toolchains/gcc.cmake -DCMAKE_C_FLAGS=\"-Werror -Wall -Wextra -Waddress -Waggregate-return -Wbad-function-cast -Wno-missing-field-initializers -Wno-cast-function-type -Wcast-align -Wformat-security -Wformat-nonliteral -Wformat=2 -Wnested-externs -Wno-unused-parameter -Wpointer-arith -Wredundant-decls  -Wundef -Wunused -Wwrite-strings -O -fno-strict-aliasing -Wuninitialized -Wno-discarded-qualifiers -Wno-incompatible-pointer-types -Wno-int-conversion -Wno-sign-conversion -Wno-pointer-sign -Wno-unused-variable\"
```

Comment:
Some of the cflags are removed, e.g. `-Wdeclaration-after-statement`, `-Wmissing-declarations`, `-Wmissing-prototypes`, others are added, e.g. `-Wno-discarded-qualifiers -Wno-incompatible-pointer-types -Wno-int-conversion -Wno-sign-conversion -Wno-pointer-sign -Wno-unused-variable`. Any reason for the change?

Reply:
A lot of the rework here was due to a number of the diagnostic flags being incompatible with the Python API. The code generated by SWIG would conflict with multiple diagnostics listed in this script (as you identified). This became more present in CMake since the SWIG file is dynamically generated a compile time, as opposed to a pre-generated file in autoconf (which the above set of flags would've been fitted too).

Though thanks for raising this as I did take an opportunity to re-evaluate this change. The diagnostics are probably still useful for the main WiredTiger C codebase, where the Python API is just a special case. As such I think the build system should be aware of what diagnostics are incompatible with the code generated by SWIG. I've updated the branch with 465c3a7e88783828a09d991476a25924d66222c5. The commit:
- Restores the original diagnostics (as there were before this branch)
- Expands the CMakeLists for the Python API to ignore specific diagnostics when compiling SWIG generated code.